src/controllers/queryRunner.ts

```diff
@@ -232,6 +245,9 @@ export default class QueryRunner {
         }
         this._statusView.executedQuery(result.ownerUri);
         let hasError = this._batchSets.some(batch => batch.hasError === true);
+        QueryRunner._activeQueries = QueryRunner._activeQueries.filter(fileName => fileName !== this.getFilePath());
```

Comment:
Filtering out the completed/errored path though is duplicated so I'd suggest moving that into its own helper function.